pyvista/core/pointset.py

```diff
@@ -818,6 +827,79 @@ def faces(self, faces):
             # TODO: faster to mutate in-place if array is same size?
             self.SetPolys(CellArray(faces))
 
+    @property
+    def regular_faces(self) -> np.ndarray:
+        """Return a face array of point indices when all faces have the same size.
+
+        Returns
+        -------
+        numpy.ndarray
+            Array of face indices with shape (n_faces, face_size).
+
+        See Also
+        --------
+        pyvista.PolyData.faces
+
+        Notes
+        -----
+        This property does not validate that the mesh's faces are all
+        actually the same size. If they're not, this property may either
+        raise a `ValueError` or silently return an incorrect array.
+
+        Examples
+        --------
+        Get the face array of a tetrahedron as a 4x3 array
+
+        >>> import pyvista as pv
+        >>> tetra = pv.Tetrahedron()
+        >>> tetra.regular_faces
+        array([[0, 1, 2],
+               [1, 3, 2],
+               [0, 2, 3],
+               [0, 3, 1]])
+
+        """
+        return _get_regular_cells(self.GetPolys())
+
+    @regular_faces.setter
+    def regular_faces(self, faces: Union[List[List[int]], np.ndarray]):
+        """Set the face cells from an (n_faces, face_size) array."""
+        self.faces = CellArray.from_regular_cells(faces, deep=True)
+
+    @classmethod
+    def from_regular_faces(cls, points, faces: Union[np.ndarray, Sequence[Sequence[int]]], deep=True):
+        """Alternate `pyvista.PolyData` convenience constructor from point and regular face arrays.
+        
+        Parameters
+        ----------
+        points : numpy.ndarray, sequence[sequence[float]]
+            A (n_points, 3) array of points.
+            
+        faces : numpy.ndarray or sequence[sequence[int]]
+            A (n_faces, face_size) array of face indices. For a triangle mesh, face_size = 3.
+
+        deep : bool, optional
+            Whether to deep copy the faces array into vtkCellArray connectivity data. Default `True`.
```

Comment:
I think our default for deep copying is usually False, but is there a reason you chose True here?  We should be consistent unless there is a reason.

Reply:
The issue here I'm not sure exactly how to handle is that you really can only set deep=False when the cells array is already a C-contiguous numpy array. If you pass in list[list[int]] it gets converted by `numpy.asarray` inside of `numpy_to_idarr`, and since nothing is holding a reference to that array it can get GC'd leaving the vtkCellArray data total garbage. My logic was to default to the safe option, and require deep=False to be opt-in. Maybe better type-checking and warning issuing would be smarter.

Reply:
Is the vtkCellArray total garbage?  Or only that you no longer have a shallow copied version?  If you deep copy, the object will also be generated inside one of the nested functions.  If it is only a problem that deep copying always occurs for lists, we can put in a note in the docstring that lists will always be deep copied.  We do this conversion to numpy array from list while claiming it is shallow copied in other locations, e.g. when setting point/cell data, so it seems strange to me that it would be a specific problem for vtkCellArray.

Reply:
This is bad indeed.

One thought is that we could add a `_set_data(offsets, connectivity)` method that stores private attributes of each, so that we maintain the references on our instance object.  I don't think this will lead to duplicating memory.

Reply:
oh yeah, that's way easier than what I was thinking